### ISSUE TITLE
OBS BROWSER OVERLAY WON'T UPDATE

### ISSUE THREAD

**Issue by al4tap** [Deep-Symmetry/beat-link-trigger#121]
I've installed everything, and the connection with my XDJ-XZ works great, so I can see all metadata... but the tracks are not updating while seen from OBS browser source. Once I do it manually, all metadata get updated on the overlay.

The expected behaviour would be that metadata track keep updated when changing the track on the players.

Beat Link Trigger Version: 0.6.3

Java Version: 11.0.9.1, OpenJDK 64-Bit Server VM,

OS : Windows 10

**Comment by brunchboy** [comment]
Hi! The OBS browser is an experimental feature, and someone will need to look at the exact code you’re using and figure out how to tweak it to get the result you want. It’s not something that GitHub is a good place to look for help for, please ask about this on the Zulip channel where user discussions are moving (Once you do, I’ll post an alert on the Gitter channel we are moving away from in case anyone who has played with the OBS stuff but hasn’t moved to Zulip yet is willing and able to help you explore this.) You can find our Zulip streams here: https://deep-symmetry.zulipchat.com/join/lp5i6yolkay7erzjzodx365q/

**Comment by brunchboy** [comment]
The [user guide](https://blt-guide.deepsymmetry.org/beat-link-trigger/0.6.3/obsoverlay#writing-template) gives you some tips about the code you will want to write:

> Don’t worry too much about the details of the templating language, because it turns out that trying to render the whole page up front doesn’t work well in the OBS browser plugin anyway. Configuring the page to refresh itself periodically, which would re-render the templates, was the original plan for this feature, but the OBS browser plugin does not use double-buffering, so that approach caused the overlay to flicker in a very unpleasant way. To get a nice-looking overlay that updates itself smoothly, write JavaScript code that periodically fetches the current player state as JSON parameters, and then updates any changed elements using DOM manipulation by ID. That is how templates, including the built-in one, are working today.

We know at least one user got a very nice updating display working this way, and he was going to share his code, but never did. So please, if and when you get this working well, please share your code, and we can update the sample and the user guide to show how you did it, making it easier for other people to get started in the future.

**Comment by brunchboy** [comment]
That said, I just tested the latest preview release of Beat Link Trigger with my XDJ-XZ, and when I started the OBS overlay server and opened a web browser on the URL that it serves, it correctly updates the metadata and waveform when I change tracks and start/stop playing. If you are seeing other things on your system, you might want to look in your browser console and see what is going on. Any errors you see there will be useful information to share with the people on Zulip. There are likely to be some people there with more OBS and front-end Javascript experience than me!

And if you find that it is working in a normal browser on your system but not in the OBS overlay server, you might be able to get more help by asking for support from the OBS community, there may be an issue with their overlay server that could be fixed, or the overlay code you write may need to be changed to accommodate limitations in that environment.
